pyMBE.py

```diff
@@ -457,6 +457,23 @@ def center_molecule_in_simulation_box(self, molecule_id, espresso_system):
                 espresso_system.part.by_id(pid).pos = es_pos - center_of_mass + box_center
         return 
 
+    def check_dimensionality(self, variable, expected_dimensionality):
+        """
+        Checks if the dimensionality of `variable` matches `expected_dimensionality`.
+
+        Args:
+            `variable`(`pint.Quantity`): Quantity to be checked.
+            `expected_dimensionality(`str`): Expected dimension of the variable.
+        """
+
+        try:
+            correct_dimensionality=variable.check(f"[{expected_dimensionality}]")
+        except:
+            raise ValueError(f"The variable {variable} should be a `pint.Quantity` object, instead the variable is {type(variable)}")
+        
+        if not correct_dimensionality:
+            raise ValueError(f"The variable {variable} should have a dimensionality of {expected_dimensionality}, instead the variable has a dimensionality of {variable.dimensionality}")
+
```

Comment:
One needs to be careful with this function, so far it works only for single dimensionality ('time', 'length' ...) but it fails when passing complex dimensionality ('length/time'), and the provided a confusing error message:

```python
pmb.check_dimensionality(1*pmb.units.nm, 'time/length')
raise ValueError(f"The variable {variable} should be a `pint.Quantity` object, instead the variable is {type(variable)}")
ValueError: The variable 1 nanometer should be a `pint.Quantity` object, instead the variable is <class 'pint.Quantity'>
```
Additionally, I misunderstood the term 'dimensionality' at my first try and I set my expected_dimensionality to 'nanometer', It raised the same confusing message error.

```python
pmb.check_dimensionality(1*pmb.units.nm, 'nanometer')
raise ValueError(f"The variable {variable} should be a `pint.Quantity` object, instead the variable is {type(variable)}")
ValueError: The variable 1 nanometer should be a `pint.Quantity` object, instead the variable is <class 'pint.Quantity'>
```
It seems that this function is primarily utilized internally. However, it may be beneficial to enhance the documentation by providing an example, such as 'length', to illustrate what dimensionality is. It should be clarified that the function operates solely on single dimensionality.

Reply:
@pinedaps thank you for pointing out this issue:
- I improved the function to work with variables with complex dimensionalities and I improved the documentation of the function to explain the expected format for `expected_dimensionality`.
-  I removed the ValueError and instead I let Pint throw its own internal error since it is actually a more clear message error.